virtualcluster/pkg/syncer/resources/pod/checker.go

```diff
@@ -170,6 +170,12 @@ func (c *controller) PatrollerDo() {
 					continue
 				}
 			}
+			// if LabelTenantSchedulerName is specified, bypass syncing
+			schedulerlabel, ok := vList.Items[i].GetLabels()[constants.LabelTenantSchedulerName]
+			if ok && schedulerlabel != "" {
+				klog.V(5).Infof("skip syncing pod with tenant scheduler label %v", schedulerlabel)
+				continue
+			}
```

Comment:
We have a func in the checker already which will filter out pods with a label rather than iterating since this is expensive - https://github.com/kubernetes-sigs/cluster-api-provider-nested/blob/e7f1762df3752d2e1206ff06b23aed6ff3c8bd40/virtualcluster/pkg/syncer/resources/pod/checker.go#L146-L150 maybe we can change this to also check for a tenant scheduler label if there is a feature-gate turning on this.

Reply:
can not use pList, since pod is not going to be synced to super cluster, so only check for vList

Reply:
Sorry ignore this, as you pointed out this is for the vPod not the pPod which that line implements.

Reply:
I see, yea my mistake, can we do that filtering here - https://github.com/kubernetes-sigs/cluster-api-provider-nested/blob/e7f1762df3752d2e1206ff06b23aed6ff3c8bd40/virtualcluster/pkg/syncer/resources/pod/checker.go#L160-L164 then.

Reply:
Which is what sets up the `differ`, so rather than loading all vPods it will filter from the informer those pods with this label (only when the feature gate is enabled)

Reply:
k